contracts/treasury/contracts/src/IZrxTreasury.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+
+interface IZrxTreasury {
+
+    struct TreasuryParameters {
+        uint256 votingPeriod;
+        uint256 proposalThreshold;
+        uint256 quorumThreshold;
+    }
+
+    struct ProposedAction {
+        address target;
+        bytes data;
+        uint256 value;
+    }
+
+    struct Proposal {
+        ProposedAction[] actions;
```

Comment:
I wonder if we should be just storing a single hash and passing the full data into `execute()` so we aren't wasting gas on storage? This is a big pain point of the current governor. They could still pass the full data into `propose()` so we have some on-chain record. And we could also emit it.

Reply:
yeah I'll do this